src/shims/env.rs

```diff
@@ -58,3 +46,81 @@ pub(crate) fn alloc_env_value<'mir, 'tcx>(
     alloc.write_bytes(&tcx, trailing_zero_ptr, &[0]).unwrap();
     ptr
 }
+
+impl<'mir, 'tcx> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+    fn getenv(
+        &mut self,
+        name_op: OpTy<'tcx, Tag>,
+        dest: PlaceTy<'tcx, Tag>
+    ) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        let result = {
+            let name_ptr = this.read_scalar(name_op)?.not_undef()?;
+            let name = this.memory().read_c_str(name_ptr)?;
+            match this.machine.env_vars.map.get(name) {
+                Some(&var) => Scalar::Ptr(var),
+                None => Scalar::ptr_null(&*this.tcx),
+            }
+        };
+        this.write_scalar(result, dest)?;
+        Ok(())
+    }
+
+    fn setenv(
+        &mut self,
+        name_op: OpTy<'tcx, Tag>,
+        value_op: OpTy<'tcx, Tag>,
+        dest: PlaceTy<'tcx, Tag>
```

Comment:
Actually `setenv` should probably return just an `i32`, and leave the `write` to the caller.

Reply:
I suppose this is also the case for `unsetenv`